hibernate-cache/src/test/java/org/infinispan/test/hibernate/cache/functional/AbstractFunctionalTest.java

```diff
@@ -239,4 +253,114 @@ protected CountDownLatch expectPutWithValue(AdvancedCache cache, Predicate<Objec
 			return new CountDownLatch(0);
 		}
 	}
+
+   protected CountDownLatch expectAfterEndInvalidation(AdvancedCache cache, int numInvalidates) {
+      CountDownLatch latch = new CountDownLatch(numInvalidates);
+      wrapInboundInvocationHandler(cache, handler -> new ExpectingInboundInvocationHandler(handler, latch));
```

Comment:
From a reader perspective, creating something within a test method and then cleaning it up in an `@AfterMethod` is asymmetric, so the user has to look somewhere else to find out where the cleanup is done... This might be necessary if there's any bugs in the clearing itself. Also, related to this, I don't know which kind of guarantees you get in terms of which thread will execute `@AfterMethod`. In theory, it should be the same thread as test method one but I've never trusted these testing frameworks. If the thread is for whatever is different, then it can get messy to figure out issues.

I agree with you that the try/finally clause is a bit clunky, but I feel it makes it clear that whatever I create in this section will be cleared there, so helps the reader IMO, and you have full guarantees of who executes the finally block.